### PR TITLE
Add method to set text and perform immediate validation.

### PR DESCRIPTION
**Pull request by wzs**
Hi, another update.
I want to set textField's text and perform immediate validation on it.

### CODE REVIEW

File: BZGFormFieldTests/BZGFormFieldTests.m [hunk 1]
```diff
@@ -124,6 +124,34 @@ - (void)testBasicTextValidation
     XCTAssertTrue([self.formField.leftIndicator.backgroundColor isEqual:self.formField.leftIndicatorValidColor]);
 }
 
+- (void)testImmidiateValidation
```
Comment (0thernet) [inline BZGFormFieldTests/BZGFormFieldTests.m]: immidiate -> immediate

`:%s/immidiate/immediate/g` ;)

File: BZGFormFieldTests/BZGFormFieldTests.m [hunk 2]
```diff
@@ -124,6 +124,34 @@ - (void)testBasicTextValidation
...
+    XCTAssertTrue([self.formField.leftIndicator.backgroundColor isEqual:self.formField.leftIndicatorNoneColor]);
+    
+    // 1 char; text set directly with immidiate validation
+    [self.formField setText:@"a" validate:YES];
+    XCTAssertTrue(self.formField.leftIndicatorState == BZGLeftIndicatorStateActive);
+    XCTAssertTrue(self.formField.leftIndicator.frame.size.width/self.formField.frame.size.height == self.formField.leftIndicatorActiveWidth);
+    XCTAssertTrue(self.formField.formFieldState == BZGFormFieldStateInvalid);
+    XCTAssertTrue([self.formField.leftIndicator.backgroundColor isEqual:self.formField.leftIndicatorInvalidColor]);
+
+        // 4 char; text set directly with immidiate validation
```
Comment (0thernet) [inline BZGFormFieldTests/BZGFormFieldTests.m]: (nitpick) indentation

File: BZGFormFieldTests/BZGFormFieldTests.m (hunk 1 above)
Comment (wzs) [inline BZGFormFieldTests/BZGFormFieldTests.m]: Fixed! But this time ort in commit msg :)

File: BZGFormField/BZGFormField.h [hunk 3]
```diff
@@ -66,6 +66,11 @@ typedef NS_ENUM(NSInteger, BZGFormFieldState) {
  */
 - (void)setAsyncTextValidationBlock:(BZGTextValidationBlock)block;
 
+/**
+ Sets text for the `textField` and perform validation.
+ */
+- (void)setText:(NSString *)text validate:(BOOL)validate;
+
```
Comment (0thernet) [inline BZGFormField/BZGFormField.h:73]: (style)

```
/**
 * Sets the textField's text and optionally performs immediate validation.
 */
- (void)setText:(NSString *)text validate:(BOOL)shouldValidate;
```

File: BZGFormField/BZGFormField.m [hunk 4]
```diff
@@ -54,6 +54,15 @@ - (void)setAsyncTextValidationBlock:(BZGTextValidationBlock)block
...
 
+- (void)setText:(NSString *)text validate:(BOOL)validate
+{
+    self.textField.text = text;
+
+    if (validate) {
+        [self.textField.delegate textFieldDidEndEditing:self.textField];
+    }
+}
+
```
Comment (0thernet) [inline BZGFormField/BZGFormField.m:65]: (style)

```
- (void)setText:(NSString *)text validate:(BOOL)shouldValidate
[...]
```

**Pull request merged**